### PR TITLE
CY-1612 - Drilldown pages don't work

### PR DESCRIPTION
**Pull request by qooban**
CY-1612 - Drilldown pages don't work after page name change or parent page name change or page removal

* Fixed page name update (drilldown page and parent page) (CY-1663)
* Improved breadcrumbs display (moved to SUIR)
* Fixed location on breadcrumb link click
* Fixed most of the ESLint problems related to updated files
* Fixed drilldown page removal (CY-1664)
* Removing children pages on page removal
* Added page reducer tests for page update (name and description) and page removal

### CODE REVIEW

File: app/components/Breadcrumbs.js [hunk 1]
```diff
@@ -3,53 +3,48 @@
...
-                        onEditDone={newName => this.props.onPageNameChange(p, newName)}
-                    />
-                );
-            }
-        });
-        return <div className="ui breadcrumb breadcrumbLineHeight">{elements}</div>;
-    }
+export default function Breadcrumbs(props) {
+    const breadcrumbElements = [];
+    const { isEditMode, onPageNameChange, onPageSelected, pagesList } = props;
```
Comment (kubama) [inline app/components/Breadcrumbs.js]: Cleaner and shorter: `export default function Breadcrumbs({ isEditMode, onPageNameChange, onPageSelected, pagesList }) {`

File: app/containers/Page.js [hunk 2]
```diff
@@ -2,56 +2,54 @@
...
-    const pagesMap = _.keyBy(pages, 'id');
+    const pagesMap = createPagesMap(pages);
     const page = pagesMap[ownProps.pageId];
     const homePageId = pages[0].id;
     const pageId = page ? page.id : homePageId;
 
     const pageData = _.clone(_.find(pages, { id: pageId }));
-    const widgets = _.map(pageData.widgets, wd => {
+    pageData.widgets = _.map(pageData.widgets, wd => {
         const w = _.clone(wd);
```
Comment (kubama) [inline app/containers/Page.js]: Although the function is very short `w` is still a nondescript name for a variable, how about `clonedWidget`?

File: app/components/Breadcrumbs.js (hunk 1 above)
Comment (qooban) [reply app/components/Breadcrumbs.js]: Agree, refactored.

File: app/containers/Page.js [hunk 3]
```diff
@@ -62,27 +60,24 @@ const mapStateToProps = (state, ownProps) => {
...
         onPageDescriptionChange: (pageId, newDescription) => {
-            dispatch(updatePageDescription(pageId, newDescription));
+            dispatch(changePageDescription(pageId, newDescription));
         },
         onPageSelected: (page, pagesList, index) => {
             const drilldownContext = [];
             // Starting from 1 cause the first page doesnt have any context and shouldnt be in the context array (only drilldown pages)
             // and also skip the last page, because we are sending the context of this one to the select page
-            for (let i = 1; i <= index - 1; ++i) {
+            for (let i = 1; i <= index - 1; i += 1) {
```
[the remark this answers is not in the document]
Comment (qooban) [reply app/containers/Page.js]: Linter reported that issue.
Ref.: https://eslint.org/docs/rules/no-plusplus

File: app/reducers/pageReducer.js [hunk 4]
```diff
@@ -55,69 +56,72 @@ const pages = (state = [], action) => {
         case types.EDIT_WIDGET:
         case types.MAXIMIZE_WIDGET:
         case types.REMOVE_WIDGET:
-            return state.map(page => {
-                if (page.id === action.pageId) {
-                    return { ...page, widgets: widgets(page.widgets, action) };
+            return state.map(p => {
```
[the remark this answers is not in the document]
Comment (qooban) [reply app/reducers/pageReducer.js]: Also here - linter reported that issue.
Ref.: https://eslint.org/docs/rules/no-shadow

**Pull request merged**